engine/src/flutter/shell/platform/windows/flutter_host_window.cc

```diff
@@ -0,0 +1,519 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/flutter_host_window.h"
+
+#include <dwmapi.h>
+
+#include "flutter/shell/platform/windows/dpi_utils.h"
+#include "flutter/shell/platform/windows/flutter_window.h"
+#include "flutter/shell/platform/windows/flutter_windows_view_controller.h"
+#include "flutter/shell/platform/windows/window_manager.h"
+
+namespace {
+
+constexpr wchar_t kWindowClassName[] = L"FLUTTER_HOST_WINDOW";
+
+// Clamps |size| to the size of the virtual screen. Both the parameter and
+// return size are in physical coordinates.
+flutter::Size ClampToVirtualScreen(flutter::Size size) {
+  double const virtual_screen_width = GetSystemMetrics(SM_CXVIRTUALSCREEN);
+  double const virtual_screen_height = GetSystemMetrics(SM_CYVIRTUALSCREEN);
+
+  return flutter::Size(std::clamp(size.width(), 0.0, virtual_screen_width),
+                       std::clamp(size.height(), 0.0, virtual_screen_height));
+}
+
+void EnableTransparentWindowBackground(
+    HWND hwnd,
+    flutter::WindowsProcTable const& proc_table) {
```

Comment:
I know that in code I've written I called used the name `proc_table`, but feel free to call this parameter just `win32` here and below. It's a shorter name and a bit more obvious. Up to you :)